include/battle.h

```diff
@@ -816,6 +816,7 @@ struct BattleStruct
     u8 quickClawRandom[MAX_BATTLERS_COUNT];
     u8 quickDrawRandom[MAX_BATTLERS_COUNT];
     u8 boosterEnergyActivates;
+    u8 distortedTypeMatchups;
```

Comment:
If you add an additional check in `MOVEEND_NEXT_TARGET` you can get away with using only one bit. 
```diff
@@ -5909,6 +5909,7 @@ static void Cmd_moveend(void)
                 gProtectStructs[gBattlerAttacker].targetAffected = TRUE;
 
             gBattleStruct->targetsDone[gBattlerAttacker] |= gBitTable[gBattlerTarget];
+            gBattleStruct->distortedTypeMatchups = 0;
             if (!(gHitMarker & HITMARKER_UNABLE_TO_USE_MOVE)
                 && gBattleTypeFlags & BATTLE_TYPE_DOUBLE
```
There is probably somewhere space for a one bit field in the struct.

Reply:
Forget what I wrote. It doesn't work though I don't understand why.